agents/agents/guard/fraud.go

```diff
@@ -0,0 +1,792 @@
+			}
+
+			return
+		})
+
+		if err != nil {
+			return fmt.Errorf("could not submit VerifyStateWithAttestation tx: %w", err)
+		}
+
+		// Submit the state report on summit.
+		srSignature, _, _, err := state.SignState(ctx, g.bondedSigner)
+		if err != nil {
+			return fmt.Errorf("could not sign state: %w", err)
+		}
+		ok, err := g.prepareStateReport(ctx, fraudAttestation.Notary, g.summitDomainID)
+		if err != nil {
+			return fmt.Errorf("could not prepare state report on summit: %w", err)
+		}
+		if !ok {
+			continue
+		}
+		_, err = g.txSubmitter.SubmitTransaction(ctx, big.NewInt(int64(g.summitDomainID)), func(transactor *bind.TransactOpts) (tx *ethTypes.Transaction, err error) {
+			tx, err = g.domains[g.summitDomainID].Inbox().SubmitStateReportWithAttestation(
+				transactor,
+				int64(stateIndex),
+				srSignature,
+				snapPayload,
+				fraudAttestation.Payload,
+				fraudAttestation.Signature,
+			)
+			if err != nil {
+				return nil, fmt.Errorf("could not submit state report with attestation on summit: %w", err)
+			}
+
+			return
+		})
+		if err != nil {
+			return fmt.Errorf("could not submit SubmitStateReportWithAttestation tx: %w", err)
+		}
+
+		// Submit the state report on the remote chain.
+		ok, err = g.prepareStateReport(ctx, fraudAttestation.Notary, fraudAttestation.AgentDomain)
+		if err != nil {
+			return fmt.Errorf("could not prepare state report on remote: %w", err)
+		}
+		if !ok {
+			continue
+		}
+		_, err = g.txSubmitter.SubmitTransaction(ctx, big.NewInt(int64(fraudAttestation.AgentDomain)), func(transactor *bind.TransactOpts) (tx *ethTypes.Transaction, err error) {
+			tx, err = g.domains[fraudAttestation.AgentDomain].LightInbox().SubmitStateReportWithAttestation(
+				transactor,
+				int64(stateIndex),
+				srSignature,
+				snapPayload,
+				fraudAttestation.Payload,
+				fraudAttestation.Signature,
+			)
+			if err != nil {
+				return nil, fmt.Errorf("could not submit state report with attestation on agent domain %d: %w", fraudAttestation.AgentDomain, err)
+			}
+
+			return
+		})
+		if err != nil {
+			return fmt.Errorf("could not submit SubmitStateReportWithAttestation tx: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// prepareStateReport checks if the given agent is in a slashable status (Active or Unstaking),
+// and relays the agent status from Summit to the given chain if necessary.
+//
+//nolint:cyclop
+func (g Guard) prepareStateReport(ctx context.Context, agent common.Address, chainID uint32) (ok bool, err error) {
+	var agentStatus types.AgentStatus
+	//nolint:nestif
+	if chainID == g.summitDomainID {
+		contractCall := func(ctx context.Context) error {
+			agentStatus, err = g.domains[chainID].BondingManager().GetAgentStatus(ctx, agent)
+			if err != nil {
+				return fmt.Errorf("could not get agent status: %w", err)
+			}
+
+			return nil
+		}
+		err = retry.WithBackoff(ctx, contractCall, g.retryConfig...)
+		if err != nil {
+			return false, fmt.Errorf("could not get agent status: %w", err)
+		}
```

Comment:
Similarly, could this be later further abstracted as `DomainClient.GetAgentStatus()`?